### PR TITLE
documentation/5337-cookies-documentation

### LINKED ISSUE #5337: Cookies - What they are and what's spawning them
## Acceptance Criteria

- [ ] List all the cookies on prod
- [ ] Document: Source, purpose, do we control it. For all cookies
- [ ] Add the table or document with this info to our Repo Readme

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #5337 

## 👋 Introduction

Adds documentation for existing cookies on the live site.

## 🕵️ Details

Read from here onward https://github.com/GCTC-NTGC/gc-digital-talent/issues/5337#issuecomment-1387470191 for more information, including various pictures/answers.
Some of these cookies may be disabled later.

## 📸 Screenshot

image

### CODE REVIEW

File: documentation/cookies.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Cookies
+
+There are various cookies present on production, ownership of cookies varies. Here are the cookies present and some details about them.
+
+# ARRAffinity
+
+This is a cookie controlled by Microsoft Azure. Its purpose is directing clients to specific instances.
```
Comment (mnigh) [inline documentation/cookies.md:7]:
```suggestion
This is a cookie controlled by Microsoft Azure. Its purpose is to direct clients to specific instances. 
```

File: documentation/cookies.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Cookies
+
+There are various cookies present on production, ownership of cookies varies. Here are the cookies present and some details about them.
+
+# ARRAffinity
+
+This is a cookie controlled by Microsoft Azure. Its purpose is directing clients to specific instances. 
+Additional [reading.](https://azure.microsoft.com/en-us/blog/disabling-arrs-instance-affinity-in-windows-azure-web-sites/)
```
Comment (mnigh) [inline documentation/cookies.md:8]:
```suggestion
[Additional information](https://azure.microsoft.com/en-us/blog/disabling-arrs-instance-affinity-in-windows-azure-web-sites/).
```

File: documentation/cookies.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# ARRAffinitySameSite
+
+This cookie is also controlled by Microsoft Azure. It's similar in purpose to ARRAffinity, but works with a relatively new SameSite browser attribute.
+
+# TS(string)
+
+This cookie is security related and is affiliated with F5 load balancers. Validation of cookies, session expiration, and ASM Frame integrity. 
+Further [reading.](https://support.f5.com/csp/article/K6850)
```
Comment (mnigh) [inline documentation/cookies.md:17]:
```suggestion
[Additional information](https://support.f5.com/csp/article/K6850).
```

File: documentation/cookies.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+There are various cookies present on production, ownership of cookies varies. Here are the cookies present and some details about them.
+
+# ARRAffinity
+
+This is a cookie controlled by Microsoft Azure. Its purpose is directing clients to specific instances. 
+Additional [reading.](https://azure.microsoft.com/en-us/blog/disabling-arrs-instance-affinity-in-windows-azure-web-sites/)
+
+# ARRAffinitySameSite
+
+This cookie is also controlled by Microsoft Azure. It's similar in purpose to ARRAffinity, but works with a relatively new SameSite browser attribute.
```
Comment (mnigh) [inline documentation/cookies.md:12]:
```suggestion
This cookie is also controlled by Microsoft Azure. It is similar in purpose to `ARRAffinity`, but works with a relatively new `SameSite` browser attribute.
```

**Review by mnigh — COMMENTED** [review]
Good start! Not sure if this meets the acceptance criteria of listing whether or not we control each cookie or not though.

**Comment by vd1992** [comment]
> Good start! Not sure if this meets the acceptance criteria of listing whether or not we control each cookie or not though.

how complete things are for this issue is a bit abstract 🤔 

I think we control or can control the Azure one, per this stack overflow answer, though can't say for sure
image


the F5 one I think probably comes from infrastructure and that's IMTD right?
control of that one 🤷

**Comment by vd1992** [comment]
the ASM thing appears to be like a firewall, and therefore should definitely be an infrastructure and security sort of thing

**Comment by petertgiles** [comment]
Yeah, F5 is a load-balancer/firewall network tool.  Same story as the process Gray is going through to tweak the admin-blocking error: Jira ticket to IMTD to request configuration changes.

**Review by petertgiles — APPROVED** [review]
Interesting read!  Thanks for the links.

**Review by mnigh — DISMISSED** [review]
👍

**Review by mnigh — CHANGES_REQUESTED** [review]
We need to list all the cookies on prod, not just ones with warnings (as per the acceptance criteria).

**Comment by vd1992** [comment]
> We need to list all the cookies on prod, not just ones with warnings (as per the acceptance criteria).

hmm yeah, true
further cookies documented here https://github.com/GCTC-NTGC/gc-digital-talent/pull/5382/commits/c5c6d23b0775f6fbde4844d26d1b75232d013faa

/admin side I cannot check

**Comment by mnigh** [comment]
> > We need to list all the cookies on prod, not just ones with warnings (as per the acceptance criteria).
> 
> hmm yeah, true, you're right further cookies documented here [c5c6d23](https://github.com/GCTC-NTGC/gc-digital-talent/commit/c5c6d23b0775f6fbde4844d26d1b75232d013faa)
> 
> /admin side I cannot check

I also cannot check the `/admin` cookies either and not sure if any additional cookies are set there anyway (not sure who has admin access to prod, @petertgiles?), though we do have some that are set after logging in on the frontend that should probably be documented:

1. gc_digital_talent_session
2. WNYQzIzpI7ZOsRdTyz9OzRnbE03WCnoEUE...
3. XSRF-TOKEN

## Screenshot
Screen Shot 2023-01-19 at 14 49 11

**Comment by vd1992** [comment]
right, I didn't want to create a prod login and pollute the user base with too many testing ones

session
XSRF
and the keyboard masher one

will look into them

**Comment by vd1992** [comment]
with some help from peter and advice, got some further documentation progress done
https://github.com/GCTC-NTGC/gc-digital-talent/pull/5382/commits/2088f79b499bb6201bfa005196170eac08019d6c and https://github.com/GCTC-NTGC/gc-digital-talent/pull/5382/commits/1fa05567195764deff5de1d2396748888c955fcb

I think the mystery cookie is figured out now, here is some further detail if you want it

image

File: documentation/cookies.md [hunk 5]
```diff
@@ -12,9 +12,38 @@ May be controlled in Azure settings at `Settings -> Configuration -> General Set
...
+Generated by logging in. 
+Owned by the web application for session logic. 
+
+## XRSF-TOKEN
+
+Generated by logging in.
+Cross Site Request Forgery token. Possibly generated by Laravel at:
+`api\vendor\laravel\framework\src\Illuminate\Foundation\Http\Middleware\VerifyCsrfToken.php`
+
+## long alphanumeric string cookie, 40 character length
```
Comment (mnigh) [inline documentation/cookies.md:38]:
```suggestion
## _long alphanumeric string_, (40 character length)
```

File: documentation/cookies.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# Cookies
+
+There are various cookies present on production, ownership of cookies varies. Here are the cookies present and some details about them.
+
+## ARRAffinity
+
+This is a cookie controlled by Microsoft Azure. Its purpose is to direct clients to specific instances. 
+[Additional information](https://azure.microsoft.com/en-us/blog/disabling-arrs-instance-affinity-in-windows-azure-web-sites/).
+May be controlled in Azure settings at `Settings -> Configuration -> General Settings -> Platform Settings`.
```
Comment (mnigh) [inline documentation/cookies.md:9]:
```suggestion
May be controlled in Azure settings at `Settings > Configuration > General Settings > Platform Settings`.
```

**Review by mnigh — COMMENTED** [review]
Thanks for adding the additional cookies 🍪.

To make it a bit easier to read and make it clear whether we "own" it or not, could you format each cookie, something like the following (I am very open to other formatting if other folks have an opinion):

## Cookie name
### Source: _Internal_ or _External_
Paragraph description of the source and purpose of the cookie.
- [Additional information](#)

**Comment by esizer** [comment]
> To make it a bit easier to read and make it clear whether we "own" it or not, could you format each cookie, something like the following (I am very open to other formatting if other folks have an opinion):

I personally like this format, makes it very clear when the cookie is internal or external and highlights the more info link 👍

**Pull request merged**